src/lib/y2partitioner/widgets/disk_expert_menu_button.rb

```diff
@@ -43,30 +48,41 @@ def label
       #
       # @return [Array<[Symbol, String]>]
       def items
-        [[:create_partition_table, _("Create New Partition &Table...")]]
+        [
+          [:create_partition_table, _("Create New Partition &Table")],
+          [:clone_disk, _("Clone this Disk")]
+        ]
       end
 
       # Event handler for the expert menu
       #
       # @param event [Hash] UI event
-      # @return [:redraw, nil] :redraw when the action is performed;
-      #   nil otherwise.
+      # @return [:redraw, nil] :redraw when the action is performed; nil otherwise.
       def handle(event)
         id = event["ID"]
         result = case id
         when :create_partition_table
           create_partition_table
+        when :clone_disk
+          clone_disk
         end
 
         result == :finish ? :redraw : nil
       end
 
     private
 
+      # @see Y2Partitioner::Actions::CreatePartitionTable
       def create_partition_table
         log.info("User selected 'create new partition table' for #{@disk.name}")
         Actions::CreatePartitionTable.new(@disk.name).run
       end
+
+      # @see Y2Partitioner::Actions::CloneDisk
+      def clone_disk
+        log.info("User selected 'clone this disk' for #{@disk.name}")
```

Comment:
just my 2c but when debugging I prefer disk.inspect over disk.name as it shows quickly important info.